### PR TITLE
[NP-310] Exit wizard after put function

### PR DESCRIPTION
**Pull request by KernelDeimos**
Note: merge #40 first to see correct diff

### CODE REVIEW

File: src/foam/nanos/crunch/ui/ScrollSectionWizardView.js [hunk 1]
```diff
@@ -121,12 +121,12 @@ foam.CLASS({
...
-        })
-        
-        x.ctrl.notify('Your progress has been saved.');
-        x.stack.back();
+        Promise.all(
+          this.sectionsList.map(wizardSection => wizardSection.save())
+        ).then(() => {
+          x.ctrl.notify('Your progress has been saved.');
+          x.stack.back();
+        });
```
Comment (Nauna) [inline src/foam/nanos/crunch/ui/ScrollSectionWizardView.js:129]: nice to have a catch statement that handles an error text.

also would be better to move the hardcoded success msg to a modeled msg.

File: src/foam/u2/crunch/CapabilityInterceptView.js [hunk 2]
```diff
@@ -95,33 +120,32 @@ foam.CLASS({
...
+      // todo find which capability was applied for
+      this.capabilityAquired = true;
+      this.capabilityOptions.forEach((c) => {
+        this.capabilityCache.set(c, true);
+      });
+      this.stack.back();
+      alert('Your permissions has changed.');
+      location.reload();
+    },
+    function reject() {
```
Comment (Nauna) [inline src/foam/u2/crunch/CapabilityInterceptView.js:134]: suggest moving this logic into the action ... don't see why it is declared in another function.

File: src/foam/u2/crunch/CapabilityInterceptView.js [hunk 3]
```diff
@@ -83,9 +87,30 @@ foam.CLASS({
...
+                            this.EQ(this.UserCapabilityJunction.SOURCE_ID, this.user.id),
+                            this.EQ(this.UserCapabilityJunction.TARGET_ID, cap.id)
+                          )).limit(1).select(this.PROJECTION(
+                            this.UserCapabilityJunction.STATUS
+                          )).then(results => {
+                            if ( results.array.length < 1 ) {
+                              view.reject();
+                              return;
+                            }
+                            var status = results.array[0][0];
```
Comment (Nauna) [inline src/foam/u2/crunch/CapabilityInterceptView.js:103]: unclear what results.array[0][0] is .. results.array[0] is th found ucj ... what is the second array access?
plz add some documentation to allow other devs to understand what is being accessed.

File: src/foam/u2/crunch/CapabilityInterceptView.js [hunk 4]
```diff
@@ -95,33 +120,32 @@ foam.CLASS({
...
+      this.capabilityAquired = true;
+      this.capabilityOptions.forEach((c) => {
+        this.capabilityCache.set(c, true);
+      });
+      this.stack.back();
+      alert('Your permissions has changed.');
+      location.reload();
+    },
+    function reject() {
+      // todo find which capability was applied for
```
Comment (Nauna) [inline src/foam/u2/crunch/CapabilityInterceptView.js:135]: is this todo still relevent?

Comment (KernelDeimos) [reply src/foam/u2/crunch/CapabilityInterceptView.js:135]: don't think so

**Comment by Nauna** [comment]
https://nanopay.atlassian.net/browse/NP-874
https://nanopay.atlassian.net/browse/NP-310

**Pull request merged**